ext/serializer.c

```diff
@@ -1060,9 +1060,12 @@ void ddtrace_serialize_span_to_array(ddtrace_span_data *span, zval *array) {
         zend_string *str_key;
         zval *val;
         ZEND_HASH_FOREACH_STR_KEY_VAL_IND(Z_ARR_P(metrics), str_key, val) {
-            if (str_key) {
+            if (str_key && strcmp(ZSTR_VAL(str_key), "process_id") != 0) {
```

Comment:
Instead of `strcmp(a, b)`, I think you need to use `zend_string_equals_literal`:

```suggestion
            if (str_key && zend_string_equals_literal(str_key, "process_id")) {
```

Same idea below as well.

Though, I wonder where the meta key is coming from, and maybe we should add its value to the `metrics` thing automatically? Meaning, if someone sets `$span->meta["process_id"] = getpid();` or whatever, that we move it at serialization time to metrics? Dunno.

Reply:
Changed the string comparison for use with a zend_string. For the meta array and possibly setting PID, do we see users ever setting that manually? Especially given that we are setting the PID within metrics no matter what. Also, the span meta is serialized during this call: `_serialize_meta(el, span);` on line 1053.